core/src/snapshot_packager_service.rs

```diff
@@ -81,7 +81,10 @@ impl SnapshotPackagerService {
                     let measure_handling = Measure::start("");
                     let snapshot_kind = snapshot_package.snapshot_kind;
                     let snapshot_slot = snapshot_package.slot;
-                    let snapshot_hash = snapshot_package.hash;
+                    let snapshot_hash = snapshot_package
+                        .snapshot_archive_package
+                        .as_ref()
```

Comment:
I know this is somewhat of a transient state, but it feels a little strange to be that we optionally have a snapshot hash here, yet require creating the snapshot archive a few lines down.

I think we should make them consistent. And this one here should `.expect()` too.